src/store/tbtc/effects.ts

```diff
@@ -78,20 +78,31 @@ export const findUtxoEffect = async (
           continue
         }
 
-        // UTXOs returned from `findAllUnspentTransactionOutputs` are in
-        // reversed order so we have to get the last element of the `utxos`
-        // array to get the oldest utxo related to this deposit address.
-        const utxo = utxos.pop()!
-
-        // Check if deposit is revealed.
-        const deposit = await forkApi.pause(
-          listenerApi.extra.threshold.tbtc.getRevealedDeposit(utxo)
-        )
+        let utxo = utxos[0]
+        let areAllDepositRevealed = true
+
+        // We have to find the first utxo that is revealed. The UTXOs returned
```

Comment:
```suggestion
        // We have to find the first utxo that is not revealed. The UTXOs returned
```

Here we are looking for the first not revealed utxo. If we find it we break the loop and redirect to step 3. So I think it should be `the first utxo that is not revealed.` 🤔

Reply:
Fixed in 30d657fc7225f883070330a10f4697567c5a9be9. I changed the comment a little bit so it describes better what's going on there (at least I think it's better, but if you don't agree then let me know in the review 😛 )